app/ts/components/Settings.tsx

```diff
@@ -27,17 +27,25 @@ export const SettingsIcon = () => {
 }
 
 export const SettingsModal = ({ display, appSettings }: { display: Signal<boolean>, appSettings: Signal<AppSettings> }) => {
-	const relayEndpointInput = useSignal({ value: appSettings.peek().relayEndpoint, valid: true })
+	const simulationRelayEndpointInput = useSignal({ value: appSettings.peek().simulationRelayEndpoint, valid: true })
+	const submitRelayEndpointInput = useSignal({ value: appSettings.peek().submitRelayEndpoint, valid: true })
 	const priorityFeeInput = useSignal({ value: formatUnits(appSettings.peek().priorityFee, 'gwei'), valid: true })
 	const blocksInFutureInput = useSignal({ value: appSettings.peek().blocksInFuture.toString(10), valid: true })
 
-	const allValidInputs = useComputed(() => relayEndpointInput.value.valid && priorityFeeInput.value.valid && blocksInFutureInput.value.valid)
+	const allValidInputs = useComputed(() => submitRelayEndpointInput.value.valid && simulationRelayEndpointInput.value.valid && priorityFeeInput.value.valid && blocksInFutureInput.value.valid)
 
-	function validateRelayEndpointInput(value: string) {
-		// https://urlregex.com/
-		const matchURL = value.match(/((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)((?:\/[\+~%\/\.\w\-_]*)?\??(?:[\-\+=&;%@\.\w_]*)#?(?:[\.\!\/\\\w]*))?)/g)
-		relayEndpointInput.value = { value, valid: !value || !matchURL || matchURL.length !== 1 }
+	// https://urlregex.com/
+	const matchUrl = (value: string) => value.match(/((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)((?:\/[\+~%\/\.\w\-_]*)?\??(?:[\-\+=&;%@\.\w_]*)#?(?:[\.\!\/\\\w]*))?)/g)
+
+	function validateSimulationRelayEndpointInput(value: string) {
+		const matched = matchUrl(value)
+		simulationRelayEndpointInput.value = { value, valid: !value || !matched || matched.length !== 1 }
+	}
+	function validateSubmitRelayEndpointInput(value: string) {
+		const matched = matchUrl(value)
+		submitRelayEndpointInput.value = { value, valid: !value || !matched || matched.length !== 1 }
 	}
```

Comment:
Thees should probably be named `set...EndpointInput`, as they have a side effect of setting state.